### ISSUE TITLE
[clang-tidy] `performance-move-const-arg` reports false positive when moving a type with a virtual destructor

### ISSUE THREAD

**Issue by sfc-gh-bhannel** [llvm/llvm-project#88434]
When I pass an rvalue reference to the implicit move constructor of a type with a virtual destructor, I get a warning for `performance-move-const-arg`. The code actually performs a move, not a copy, and the warning is incorrect. Removing `std::move` would cause a copy where previously none occurred.

Minimal reproduction:
```
#include <utility>

struct S {
  virtual ~S() = default;
};

int main() {
  S s1;
  S s2(std::move(s1));
  return 0;
}
```

```
$ clang-tidy test.cpp
113 warnings generated.
test.cpp:9:8: error: passing result of std::move() as a const reference argument; no move will actually happen [performance-move-const-arg,-warnings-as-errors]
  S s2(std::move(s1));
       ^~~~~~~~~~  ~
Suppressed 113 warnings (112 in non-user code, 1 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
1 warning treated as error
$ clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 16.0.0
  Optimized build.
```

Reproduction in compiler explorer: https://godbolt.org/z/Excbed74j

**Comment by sfc-gh-bhannel** [comment]
Sorry, I am incorrect. The explicitly declared virtual destructor (even though it is defaulted) prevents implicit generation of a move constructor.
https://en.cppreference.com/w/cpp/language/move_constructor
